src/main/java/edu/ksu/canvas/interfaces/RolesReader.java

```diff
@@ -0,0 +1,17 @@
+package edu.ksu.canvas.interfaces;
+
+import edu.ksu.canvas.model.Role;
+import edu.ksu.canvas.requestOptions.ListRolesOptions;
+
+import java.io.IOException;
+import java.util.List;
+
+public interface RolesReader extends CanvasReader<Role, RolesReader> {
```

Comment:
To match the naming of the rest of the interfaces in this package, this should probably be singular like `RoleReader` instead of `RolesReader`. I guess this applies to the `RolesImpl` class too (although I see the existing `SectionsImpl` class doesn't follow this convention)